src/librbd/io/ObjectDispatcher.cc

```diff
@@ -179,17 +181,19 @@ int ObjectDispatcher<I>::prepare_copyup(
   auto cct = this->m_image_ctx->cct;
   ldout(cct, 20) << "object_no=" << object_no << dendl;
 
-  std::shared_lock locker{this->m_lock};
+  this->m_op_tracker->start_op();
```

Comment:
Would be nice to use RAII here so that the clean up happens automatically when a local variable goes out of scope. See https://stackoverflow.com/questions/14276508/how-does-scope-locking-work. In fact you probably only need to hold the lock in m_op_tracker rather than starting+ending an op